stm32 rtc driver using the stm32 LL_RTC Time Date init functions

FRASTM opened the pull request:
Use the LL_RTC_TIME_Init and LL_RTC_DATE_Init and 
let the LL functions manage the Enter init mode and the Write Protection of the RTC calendar Registers

Increase the Timeout when executing the tests/drivers/rtc/rtc_api on stm32 targets
This will let test ends correctly when waiting for the alarm flags

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/72888

FRASTM committed:
tests: drivers: rtc api testing with timeout

Add a timeout to the tests/drivers/rtc_api testcase so twister
can end properly when running with twister.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
drivers: rtc: stm32 rtc driver configure time and Date

Set a Calendar using the LL functions LL_RTC_TIME_Init
and LL_RTC_DATE_Init. Add more DBG info
On the stm32F2 serie, check RS Flag after setting calendar registers.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

ajarmouni-st commented on drivers/rtc/rtc_ll_stm32.c:
In `rtc_stm32_alarm_set_time()`
```
	if ((mask == 0) && (timeptr == NULL)) {
		memset(&(p_rtc_alrm->ll_rtc_alrm), 0, sizeof(LL_RTC_AlarmTypeDef));
		p_rtc_alrm->user_callback = NULL;
		p_rtc_alrm->user_data = NULL;
		p_rtc_alrm->is_pending = false;
#if RTC_STM32_BACKUP_DOMAIN_WRITE_PROTECTION
		LL_PWR_EnableBkUpAccess();
#endif /* RTC_STM32_BACKUP_DOMAIN_WRITE_PROTECTION */
		if (rtc_stm32_is_active_alarm(RTC, id)) {
			LL_RTC_DisableWriteProtection(RTC);
			rtc_stm32_disable_alarm(RTC, id);
+                       rtc_stm32_disable_interrupt_alarm(RTC, id);
			LL_RTC_EnableWriteProtection(RTC);
		}
		LOG_DBG("Alarm %d has been disabled", id);
		goto disable_bkup_access;
	}
```

FRASTM replied:
`rtc_stm32_disable_interrupt_alarm` now added

ajarmouni-st commented on drivers/rtc/rtc_ll_stm32.c:
```diff
@@ -187,12 +187,29 @@ static inline ErrorStatus rtc_stm32_init_alarm(RTC_TypeDef *rtc, uint32_t format
 					LL_RTC_AlarmTypeDef *ll_alarm_struct, uint16_t id)
 {
 	ll_alarm_struct->AlarmDateWeekDaySel = RTC_STM32_ALRM_DATEWEEKDAYSEL_DATE;
-	/* RTC write protection is disabled & enabled again inside LL_RTC_ALMx_Init functions */
+	ll_alarm_struct->AlarmDateWeekDay = 0x01;
```
These lines are uncessary & potentially incorrect since the `AlarmDateWeekDaySel` & `AlarmDateWeekDay` fields of `LL_RTC_AlarmTypeDef` are initialized inside `rtc_stm32_init_ll_alrm_struct()`

FRASTM replied:
removed

ajarmouni-st reviewed (changes requested):
can you please put the modifs related to RTC alarm in a separate commit?

FRASTM committed:
drivers: rtc: stm32 rtc driver disable alarm interrupt if not used

Note that the Init of the alarm has no need for BCD2BIN conversion,
the LL_RTC_ALMx_Init will do. Add more log debug
The driver must clear the ALARM enable bit before wrting the alarm
registers.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
rebase on ec0e2d6308680203bb58275062b79105016ea1b1
